Sources/SwiftFormat/Parsing.swift

```diff
@@ -40,15 +40,19 @@ func parseAndEmitDiagnostics(
   let sourceFile =
     try operatorTable.foldAll(Parser.parse(source: source)) { _ in }.as(SourceFileSyntax.self)!
 
+  let diagnostics = ParseDiagnosticsGenerator.diagnostics(for: sourceFile)
   if let parsingDiagnosticHandler = parsingDiagnosticHandler {
     let expectedConverter =
       SourceLocationConverter(file: url?.path ?? "<unknown>", tree: sourceFile)
-    let diagnostics = ParseDiagnosticsGenerator.diagnostics(for: sourceFile)
     for diagnostic in diagnostics {
       let location = diagnostic.location(converter: expectedConverter)
       parsingDiagnosticHandler(diagnostic, location)
     }
   }
 
+  guard diagnostics.isEmpty else {
+    throw SwiftFormatError.fileContainsInvalidSyntax
```

Comment:
I wonder if we should eventually ditch the `parsingDiagnosticsConsumer` model and just throw the array of diagnostics as part of this error, since we don't plan on doing anything else after that.

That's a bigger API change though so we can save it for later, to make sure there aren't any other weird side effects.